src/components/pages/PageDecksEdit/PageDecksEdit.jsx

```diff
@@ -0,0 +1,237 @@
+  );
+
+  const populateData = useCallback(
+    async () => {
+      const characters = await populateCharacters();
+      const circumstances = await populateCircumstances();
+      const conflicts = await populateConflicts();
+
+      await Promise.all([characters, circumstances, conflicts])
+        .then((values) => values);
+
+      const initialDeckData = await firebaseCollectionsHelper
+        .getDeckData(id);
+      setDeck(initialDeckData);
+
+      const selectedCharacters = initialDeckData.characterCards
+        .map((card) => characters.find((character) => (
+          character.value === card.cardRef.id
+        )))
+        .filter((i) => i);
+      setSelectedCharacterIds(selectedCharacters);
+
+      const selectedCircumstances = initialDeckData.circumstanceCards
+        .map((card) => circumstances.find((circumstance) => (
+          circumstance.value === card.cardRef.id
+        )))
+        .filter((i) => i);
+      setSelectedCircumstanceIds(selectedCircumstances);
+
+      const selectedConflicts = initialDeckData.conflictCards
+        .map((card) => conflicts.find((conflict) => (
+          conflict.value === card.cardRef.id
+        )))
+        .filter((i) => i);
+      setSelectedConflictIds(selectedConflicts);
+
+      setIsLoading(true);
+    },
+    [id, populateCharacters, populateCircumstances, populateConflicts],
+  );
+
+  function changeDeck(key, value) {
+    const newDeck = { ...deck };
+    newDeck[key] = value;
+    setDeck(newDeck);
+  }
+
+  const handleUpdate = useCallback(
+    async () => {
+      const newDeck = {
+        ...deck,
+        name: deck.name,
+        description: deck.description,
+        characterCards: selectedCharacterIds.map((cardId) => (
+          {
+            cardRef: firebaseCollectionsHelper.getCardRef(cardId.value),
+            quantity: 1,
+          }
+        )),
+        circumstanceCards: selectedCircumstanceIds.map((cardId) => (
+          {
+            cardRef: firebaseCollectionsHelper.getCardRef(cardId.value),
+            quantity: 1,
+          }
+        )),
+        conflictCards: selectedConflictIds.map((cardId) => (
+          {
+            cardRef: firebaseCollectionsHelper.getCardRef(cardId.value),
+            quantity: 1,
+          }
+        )),
+      };
+      console.log('newDeck', newDeck, id);
+      firebaseCollectionsHelper.updateDeck(id, newDeck);
+    },
+    [id, deck, selectedCharacterIds, selectedCircumstanceIds, selectedConflictIds],
+  );
+
+  useEffect(() => {
+    populateData();
+  }, [populateData]);
+
+  return (
+    <TemplateDefault data-test="p-decks-edit">
+      <h1>Edit Deck</h1>
+      <If
+        condition={!isLoading}
+        data-test="p-decks-edit__loading"
+      >
+        <LoadingAnim />
+
+        <Else>
+          <form
+            data-test="p-decks-edit__form"
+            onSubmit={(e) => {
+              e.preventDefault();
+              handleUpdate();
+              history.push('/decks');
```

Comment:
You need to wait to change the route.  You should only do so when the update is complete successfully